### PR TITLE
Adding average

### LINKED ISSUE #5: Add Average method
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.average?view=netcore-2.1

### PR DESCRIPTION
**Pull request by jmeline**
closes #5

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -3,4 +3,5 @@ interface Array<T> {
   Aggregate<T>(accum: (memo: T, val: T) => T): T;
   Select<T, U>(xform: (x: T) => U): Array<U>;
   Where<T>(predicate: (X: T) => boolean): Array<T>;
+  Average<number>(): number;
```
Comment (jreina) [inline index.d.ts]: There is another overload of `Average<T>((x: T) => number): number`. We can merge as-is and leave the issue open. Do you want to merge this as-is?

File: src/Average.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/**
+ * @param None
```
Comment (jreina) [inline src/Average.js]: This `@param` annotation can be omitted entirely unless defining the overload which takes a function here.

File: test/Average.spec.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+describe('Array#prototype#Average', function() {
+  it('Should return the first 4 numbers', function() {
+    const input = [2, 4, 6, 8];
+    const expected = 5;
+    const actual = input.Average();
+    expect(actual).to.eql(expected);
+  });
+
+  it('Should throw an error when source is null', function() {
```
Comment (jreina) [inline test/Average.spec.js]: This is actually testing to see if it throws when the sequence contains no elements 🙂

Comment (jmeline) [reply test/Average.spec.js]: Renamed test to be more accurate :)

File: src/Average.js (hunk 2 above)
Comment (jmeline) [reply src/Average.js]: I implemented the overload case and fixed this annotation to correctly match that overload.

File: index.d.ts (hunk 1 above)
Comment (jmeline) [reply index.d.ts]: I've covered both cases now in my latest commit.

**Comment by jmeline** [comment]
I merged with your develop branch. I hope that was alright. It should cleanly merge now. Let me know if there is anything else you want me to do in this PR. Thanks!

**Comment by jreina** [comment]
Thank you for your contribution. Happy hacking 🎃

**Pull request merged**